code/modules/surgery/organs/augments_arms.dm

```diff
@@ -54,8 +54,17 @@
 	to_chat(user, "<span class='notice'>You modify [src] to be installed on the [zone == BODY_ZONE_R_ARM ? "right" : "left"] arm.</span>")
 	update_icon()
 
+/obj/item/organ/cyberimp/arm/Insert(mob/living/carbon/M, special = FALSE, drop_if_replaced = TRUE)
+	. = ..()
+	var/side = zone == BODY_ZONE_R_ARM? RIGHT_HANDS : LEFT_HANDS
+	hand = owner.hand_bodyparts[side]
+	if(hand)
+		RegisterSignal(hand, COMSIG_ITEM_ATTACK_SELF, .proc/ui_action_click) //If the limb gets an attack-self, open the menu. Only happens when hand is empty
+
 /obj/item/organ/cyberimp/arm/Remove(mob/living/carbon/M, special = 0)
 	Retract()
+	if(hand)
```

Comment:
What happens if the owner gains or loses a hand in between these two events?

Reply:
If the arm that the implant is meant for gets lost, the implant gets removed from the human and stored inside the arm. It's a bit wacky that it's not in the arm by default but I assume it has something to do with how implant code works.

Putting the lost arm back on the human re-adds the implant, and no functionality in this PR breaks from any of that.